tests/Unit/Presentation/SofortUrlGeneratorTest.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Tests\Unit\Presentation;
+
+use PHPUnit\Framework\TestCase;
+use Sofort\SofortLib\Sofortueberweisung;
+use WMDE\Euro\Euro;
+use WMDE\Fundraising\Frontend\Presentation\SofortUrlConfig;
+use WMDE\Fundraising\Frontend\Presentation\SofortUrlGenerator;
+
+class SofortUrlGeneratorTest extends TestCase {
+
+	public function testGenerateUrlSuccess(): void {
+		$config = SofortUrlConfig::newFromConfig( [
+			'config-key' => 'fff',
+			'item-name' => 'Donation',
+			'return-url' => 'https://us.org/yes',
+			'cancel-url' => 'https://us.org/no'
+		] );
+
+		$api = $this->createMock( Sofortueberweisung::class );
+		$api
+			->expects( $this->once() )
+			->method( 'setAmount' )
+			->with( '5.00' )
+			->willReturnSelf();
```

Comment:
This prompted me to pull this code and have a look at `Sofortueberweisung`, because it's weird to mock out non-services. Looking at the code a bit it seems reasonable enough here though.

I'm wondering if doing this via subclassing would not be clearer.

I'm also wondering if the `->expects( $this->once() )` make sense. Do you need to verify these methods only get called a specific amount of times? (See point 3 in https://www.entropywins.wtf/blog/2016/05/13/5-ways-to-write-better-mocks/)